lib/xcodeproj/scheme/abstract_scheme_action.rb

```diff
@@ -23,6 +23,102 @@ def build_configuration
+      #         Each entry can be either a 'Run Script' or a 'Send Email' action.
+      #
+      def pre_actions=(pre_actions)
+        return if pre_and_post_actions_disabled
+
+        @xml_element.delete_element('PreActions')
+        unless pre_actions.empty?
+          pre_actions_element = @xml_element.add_element('PreActions')
+          pre_actions.each do |entry_node|
+            pre_actions_element.add_element(entry_node.xml_element)
+          end
+        end
+        pre_actions
+      end
+
+      # @param [ExecutionAction] pre_action
+      #        Add an action to the list of actions to run before this scheme action.
+      #        It can be either a 'Run Script' or a 'Send Email' action.
+      #
+      def add_pre_action(pre_action)
+        return if pre_and_post_actions_disabled
+
+        pre_actions = @xml_element.elements['PreActions'] || @xml_element.add_element('PreActions')
+        pre_actions.add_element(pre_action.xml_element)
+      end
+
+      # @return [Array<ExecutionAction>]
+      #         The list of actions to run after this scheme action.
+      #         Each entry can be either a 'Run Script' or a 'Send Email' action.
+      #
+      def post_actions
+        return if pre_and_post_actions_disabled
+
+        post_actions = @xml_element.elements['PostActions']
+        return nil unless post_actions
+        post_actions.get_elements('ExecutionAction').map do |entry_node|
+          ExecutionAction.new(nil, entry_node)
+        end
+      end
+
+      # @param [Array<ExecutionAction>] post_actions
+      #        Set the list of actions to run after this scheme action.
+      #        Each entry can be either a 'Run Script' or a 'Send Email' action.
+      #
+      def post_actions=(post_actions)
+        return if pre_and_post_actions_disabled
+
+        @xml_element.delete_element('PostActions')
+        unless post_actions.empty?
+          post_actions_element = @xml_element.add_element('PostActions')
+          post_actions.each do |entry_node|
+            post_actions_element.add_element(entry_node.xml_element)
+          end
+        end
+        post_actions
+      end
+
+      # @param [ExecutionAction] post_action
+      #        Add an action to the list of actions to run after this scheme action.
+      #        It can be either a 'Run Script' or a 'Send Email' action.
+      #
+      def add_post_action(post_action)
+        return if pre_and_post_actions_disabled
+
+        post_actions = @xml_element.elements['PostActions'] || @xml_element.add_element('PostActions')
+        post_actions.add_element(post_action.xml_element)
+      end
+
+      #-------------------------------------------------------------------------#
+
+      private
+
+      # @!group Private helpers
+
+      # @return [Bool]
+      #         The AnalyzeAction does not support pre/post actions (see in Xcode)
+      def pre_and_post_actions_disabled
```

Comment:
nits: add an empty line after the ruby doc and the method name.

This one can be renamed to `pre_and_post_actions_disabled?`